packages/framework/src/Component/Image/ImageFacade.php

```diff
@@ -329,9 +339,9 @@ public function copyImages($sourceEntity, $targetEntity)
         $sourceImages = $this->getAllImagesByEntity($sourceEntity);
         $targetImages = [];
         foreach ($sourceImages as $sourceImage) {
-            $this->filesystem->copy(
-                $this->imageLocator->getAbsoluteImageFilepath($sourceImage, ImageConfig::ORIGINAL_SIZE_NAME),
-                $this->fileUpload->getTemporaryFilepath($sourceImage->getFilename())
+            $this->mountManager->copy(
+                'main://' . $this->imageLocator->getAbsoluteImageFilepath($sourceImage, ImageConfig::ORIGINAL_SIZE_NAME),
+                'local://' . $this->fileUpload->getTemporaryFilepath($sourceImage->getFilename())
```

Comment:
I'm really confused about this solution. It seems inappropriate to set here wrapper directly. 
Do you need to copy the images from the main filesystem (i believe this could be for example Google Cloud Bucket) to the local filesystem (disk)?

And how does the inconsistency between `ImageFacade` and `FileUpload` happened?

Reply:
inconsistency happened via implementation of `Flysystem` 
as we can see here the first step of the copying 
```
$this->filesystem->copy(
                $this->imageLocator->getAbsoluteImageFilepath($sourceImage, ImageConfig::ORIGINAL_SIZE_NAME),
                $this->fileUpload->getTemporaryFilepath($sourceImage->getFilename())
``` tries to copy for example gc://product/1.jpg into gc://%kernel.root_dir%/var/cache/dev/fileUploads and the second `FileUpload.php` class
```
$this->mountManager->move('local://' . $sourceFilepath, 'main://' . $targetFilename);
```
tries to copy it from %kernel.root_dir%/var/cache/dev/fileUploads so this PR fixes the first part where the copy is from GC into local container filesystem and the second part could copy it into GC with the new id.png or id.jpg that it will get from Doctrine eventlistener postPersist.

Reply:
so old buggy way was GC -> GC Local(could not be here since it was copied into GC) -> GC
new way is GC -> Local Local -> GC

Reply:
Okay, I'm still not convinced, using wrappers directly is a good idea, but it's consistent with other usages (in `FileUpload` or `FeedExport`).
So it's fine by me.

Reply:
ooooo, Feed export, daaamn. Another temporary path in external storage. @TomasLudvik is that right? I will create some issue for this

Reply:
oh it seems to be OK. $feed->getTemporaryFilepath() just adds suffix `.tmp` so no var/www/html/var/cache is created